File: VB/FilteredListViewNavigationControl.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.BaseImpl

Namespace FilteredListViewNavigationControl.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			Dim task1 As Task = ObjectSpace.FindObject(Of Task)(CriteriaOperator.Parse("Subject == 'Fix breakfast'"))
			If task1 Is Nothing Then
				task1 = ObjectSpace.CreateObject(Of Task)()
				task1.Subject = "Fix breakfast"
				task1.StartDate = DateTime.Today.AddDays(-1)
				task1.DueDate = DateTime.Today
				task1.Status = DevExpress.Persistent.Base.General.TaskStatus.Completed
			End If
			Dim task2 As Task = ObjectSpace.FindObject(Of Task)(CriteriaOperator.Parse("Subject == 'Review reports'"))
			If task2 Is Nothing Then
				task2 = ObjectSpace.CreateObject(Of Task)()
				task2.Subject = "Review reports"
				task2.StartDate = DateTime.Today
				task2.DueDate = DateTime.Today.AddDays(1)
				task2.Status = DevExpress.Persistent.Base.General.TaskStatus.InProgress
			End If
		End Sub
	End Class
End Namespace

File: CS/FilteredListViewNavigationControl.Module/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp;
using DevExpress.Persistent.BaseImpl;

namespace FilteredListViewNavigationControl.Module {
	public class Updater : ModuleUpdater {
		public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) {
		}
		public override void UpdateDatabaseAfterUpdateSchema() {
			base.UpdateDatabaseAfterUpdateSchema();
            Task task1 = ObjectSpace.FindObject<Task>(
            CriteriaOperator.Parse("Subject == 'Fix breakfast'"));
            if (task1 == null) {
                task1 = ObjectSpace.CreateObject<Task>();
                task1.Subject = "Fix breakfast";
                task1.StartDate = DateTime.Today.AddDays(-1);
                task1.DueDate = DateTime.Today;
                task1.Status = DevExpress.Persistent.Base.General.TaskStatus.Completed;
            }
            Task task2 = ObjectSpace.FindObject<Task>(
            CriteriaOperator.Parse("Subject == 'Review reports'"));
            if (task2 == null) {
                task2 = ObjectSpace.CreateObject<Task>();
                task2.Subject = "Review reports";
                task2.StartDate = DateTime.Today;
                task2.DueDate = DateTime.Today.AddDays(1);
                task2.Status = DevExpress.Persistent.Base.General.TaskStatus.InProgress;
            }
		}
	}
}
